packages/core/src/Badge/Badge.js

```diff
@@ -71,8 +71,8 @@ const Badge = styled.div`
   text-transform: uppercase;
   letter-spacing: ${themeGet('letterSpacings.caps')};
   ${space} ${type} ${color};
-  ${applySizes(sizes)}
-  ${applyVariations('Badge')}
+  ${({ theme }) => applySizes(sizes, undefined, theme.mediaQueries)};
```

Comment:
This ensure that the sizes are true to the current theme in context and not the static sizes from the DS theme.